[rcore] cmake for platform SDL is broken

Please, before submitting a new issue verify and check:

 - [X] I tested it on latest raylib version from master branch
 - [X] I checked there is no similar issue already reported
 - [X] I checked the documentation on the [wiki](https://github.com/raysan5/raylib/wiki)
 - [x] My code has no errors or misuse of raylib

### Issue description

I am trying to compile raylib master with the SDL3 platform on Windows 11. 
- I have downloaded SDL3 (SDL3-devel-3.2.24-mingw.zip), unzipped that folder and added the following environment variable: CMAKE_PREFIX_PATH with the path to the "cmake" subfolder of the unzipped SDL3 folder.
- I have downloaded the zipped folder of the raylib master branch, unzipped the folder, created a new build folder in it, opened a PowerShell window in that build folder and entered the following:
1- cmake .. -G "MinGW Makefiles" -DPLATFORM=SDL
2- cmake --build .

I get the following error message when building (second step):

> raylib-master\src\rcore.c:2343:44: error: 'MAX_PATH' undeclared (first use in this function)
> 2343 len = GetModuleFileNameA(NULL, appDir, MAX_PATH);

I am able to build raylib master on the GLFW platform without problems using the following commands: 
1- cmake .. -G "MinGW Makefiles" -DPLATFORM=Desktop.
2- cmake --build .

### Environment

Platform: SDL (SDL3) vs GLFW ,
Operating System: Windows 11
OpenGL version: 3.3.0
GPU: Intel(R) UHD Graphics 620

### Issue Screenshot

I used to be able to compile SDL, and upon looking at the commit history of rcore.c, I see that the definition of MAX_PATH has been removed in commit number 7191749: 

<img width="1496" height="424" alt="Image" src="https://github.com/user-attachments/assets/1662a93d-f3da-4052-bfce-723f6f8fb03b" />

As it must have been removed for a reason, and as I do not know why the value of 1025 was assigned to MAX_PATH, I do not feel comfortable simply restoring it to the way it was.

--- a/src/rcore.c
+++ b/src/rcore.c
@@ -26,7 +26,8 @@
 *           - Linux DRM subsystem (KMS mode)
 *       > PLATFORM_ANDROID:
 *           - Android (ARM, ARM64)
-*
+*       > PLATFORM_DESKTOP_WIN32 (Native Win32):
+*           - Windows (Win32, Win64)
 *   CONFIGURATION:
 *       #define SUPPORT_DEFAULT_FONT (default)
 *           Default font is loaded on window initialization to be available for the user to render simple text.
@@ -161,7 +162,10 @@
 #endif
 
 // Platform specific defines to handle GetApplicationDirectory()
-#if (defined(_WIN32) && !defined(PLATFORM_DESKTOP_RGFW)) || (defined(_MSC_VER) && defined(PLATFORM_DESKTOP_RGFW))
+#if defined(_WIN32)
+    #if !defined(MAX_PATH)
+        #define MAX_PATH 260
+    #endif
 
 struct HINSTANCE__;
 #if defined(__cplusplus)

